diff --git a/node_modules/expo-location/android/build.gradle b/node_modules/expo-location/android/build.gradle
index 190f8b4..aaacfe0 100644
--- a/node_modules/expo-location/android/build.gradle
+++ b/node_modules/expo-location/android/build.gradle
@@ -82,7 +82,7 @@ android {
 dependencies {
   implementation project(':expo-modules-core')
 
-  api 'com.google.android.gms:play-services-location:20.0.0'
+  api 'com.google.android.gms:play-services-location:21.0.1'
 
   api('io.nlopez.smartlocation:library:3.3.3') {
     transitive = false
